### ISSUE TITLE
Roll20 Compendium for Open Legend

### ISSUE THREAD

**Issue by frankieali** [openlegend/core-rules#479]
Hey Brian,
I'd like to create a compendium for roll20. I'm a seasoned developer with plenty of knowledge in JSON data structures. I've already started some data transforms on the yaml data to see how well things would convert. Would you be willing to collaborate with me on this effort? I think getting a compendium into roll20 would really help to promote the platform overall.

**Comment by frankieali** [comment]
A compendium in roll20 is a collection of all the game rules made available by an Open Game License (OGL). Roll20 currently has compendiums available for [AD&D 5e](https://roll20.net/compendium/dnd5e/BookIndex), [Pathfinder](https://roll20.net/compendium/pathfinder/BookIndex) and [Dungeon World](https://roll20.net/compendium/dw/BookIndex).

The compendium adds features to the roll20 application, including Character Sheet integration and drag and drop functionality. There is a [guild for building new compendiums](https://wiki.roll20.net/Building_Compendiums). There's also been an [open call for contributors](https://app.roll20.net/forum/post/4314710/help-us-create-more-compendium-options). I think Open Legend would be a perfect addition.

Once a compendium has been added, all the Open Legend rules become available for search within the roll20 game itself. I think this would really help lower the barrier for entry into a new system for a lot of players.

New compendiums can be added in the future. For instance, Wizards of the Coast have the core rules available for free, but the [Monster Manual](https://marketplace.roll20.net/browse/bundle/3/monster-manual/) is a purchasable addon from the Marketplace.

The only catch is 

> one Editor per game system Compendium

Really, I view that as one Compendium Admin, and the assets themselves could live on Github for collaboration and feedback.
I'll reach out to @BuildAndDie as well concerning this idea. I'd really like to see it happen.

**Comment by BuildAndDie** [comment]
Yes, I've always been planning on working on the compendium eventually. I need to make the actually Roll20 Charactersheet compendium compatiable for it to work completely.

Part of it is having the Developer SRD finished though as well, b/c the SRD will be what is allowed to be included in the compendium. Currently the SRD is waiting on the kickstarter goals to be finished first I believe. I think the only real difference between what is on the Website/Github and what will be in the SRD are mainly some (or all) of the "description" text @brianfeister ?

But, I would be happy to help some with it. I travel a lot, so I usually work on the sheet in bursts. I just got the sheet up to the point now were I'm ready to do a complete overhaul on the coding (clean up and make more efficient, especially on the CSS and Workerscripts). When I do that, I was planning on including things for both the way Roll20 does for translations and for Compendium.

Of course the compendium can be used regardless without it being tied to a charactersheet, but having the addition of being able to drag and drop a boon or bane onto the sheet would be super nice.

**Comment by BuildAndDie** [comment]
Also, whereas it is one editor, I believe you can have others that help, that was part of the call out they did awhile back, to get people to help with data entry? It's been.... many many months since I looked at all that though, so it could have changed slightly, or my memory is just a little fuzzy on the details.

**Comment by frankieali** [comment]
I've already started transforming some of the data just to see what the level of effort would be. The yaml in the SRD was a great starting point, but I've had to reformat the objects a bit to better comply with roll20's requirements. There are also some places where the data gets mixed in a bit too much, such as power level tiers within the effect description. The data becomes more useful when the power levels are broken out into their own category. I'll throw up a repo into Github later this week once I've done some clean up. The way the JSON is formatted, it wouldn't be hard to throw it all into a Mongo DB to test drive some view templates.
I'll also take a closer look at your character sheet to see what kind of field names you're using. We'll want those to map to the key values in the compendium data.
I've reached out to the roll20 team as well concerning where it's possible to use markdown and where it's possible to use HTML. It's not clear from scraping the D&D compendium pages what is used where. It is obvious that they're doing some view level JavaScript processing and not just printing out the raw JSON values.

Is there an SRD planned for Amaurea's Dawn?

**Comment by frankieali** [comment]
I just got a message back from Miles P. at Roll20 concerning compendium development.
>Hi Frankie Ali,
>Thank you for your interest in helping with our Compendium - We are not currently onboarding editors at this time.
>
>However, we have added your name to our list and we will reach out when we are ready to start working on additional compendiums!
>
>Regards,
>Miles

So it looks like the compendium will mostly be developed offline for now. It seems like Roll20 won't be able to provide support concerning their JSON structure and parsing capabilities. I'll continue to build out my JSON api's and see if I can get some mock views set up as well.
Character sheet integration will have to wait until we get the green light to upload the compendium.

**Comment by BuildAndDie** [comment]
Sounds like a plan. They might be slow to accept compendiums unfortunately, right now I think they only have 5e and pathfinder?

But I'll still end up updating all my fields and stuff to be more compendium friendly on the sheet for when that time comes.

**Comment by frankieali** [comment]
I've uploaded my initial go at the Compendium. It's available here: https://github.com/frankieali/open-legend-compendium. Please shoot me over any feedback you have. I'm going to work on getting through the content pages next.

**Comment by frankieali** [comment]
I did a big refactor today. I think I'm in a good spot to start building out some mock files and view templates.

**Comment by BuildAndDie** [comment]
sounds good. Hopefully Roll20 will open up compendiums soon to more games so we can put it into play, though it would be nice to get a dice rolling type specific to Open Legend's way of rolling explosions with advantage.

**Comment by frankieali** [comment]
I've added a site for displaying the compendium data https://open-legend-grimoire.herokuapp.com/
